updateLeantime.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+
+#Declaring used colors
+BLUE='\033[1;34m'
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m'
+
+#Declaring used functions
+getConfigVar () {
+	grep "\$$1" config/configuration.php |awk -F "=" '{print $2}' |awk -F ";" '{print $1}' |tr -d ' ' |tr -d "\'"
+}
+
+getDateString () {
+	date '+%Y%m%d_%H%M%S'
+}
+
+echo -e "${BLUE}"
+echo "    _                 _   _                             "
+echo "   | |   ___ __ _ _ _| |_(_)_ __  ___                   "
+echo "   | |__/ -_) _\` | ' \  _| | '  \/ -_)                  "
+echo "   |____\___\__,_|_||_\__|_|_|_|_\___|      _           "
+echo "                     | | | |_ __  __| |__ _| |_ ___ _ _ "
+echo "                     | |_| | '_ \/ _\` / _\` |  _/ -_) '_|"
+echo "                      \___/| .__/\__,_\__,_|\__\___|_|  "
+echo "                           |_|                          "
+echo -e "${NC}"
+
+echo -e "\nFor correct operation of this script, please ensure you have zip, unzip, wget and curl installed.\n"
+
+CURRENT_VERSION=$(grep "appVersion" config/appSettings.php |awk -F' = ' '{print "v"substr($2,2,length($2)-3)}')
+LATEST_RELEASE=$(curl -L -s -H 'Accept: application/json' https://github.com/leantime/leantime/releases/latest)
+LATEST_VERSION=$(echo $LATEST_RELEASE | sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
+DOWNLOAD_URL=$(echo "https://github.com/leantime/leantime/releases/download/$LATEST_VERSION/Leantime-$LATEST_VERSION.zip")
+
+
+
+if [[ "$CURRENT_VERSION" == "$LATEST_VERSION" ]]
+then
+	#No update available
+	echo -e "${GREEN}You are already up to date with $LATEST_VERSION!${NC}"
+else
+	#Update available
+	echo -e "${RED}There is an update available!${NC}"
+	read -r -p "Do you want to update from $CURRENT_VERSION to $LATEST_VERSION? [Y/n] " chUpd
+	case $chUpd in
+		Y|y|Yes|yes|"")
+			####################
+                        # BACKUP           #
+			####################
+			read -r -p "Do you want to create a backup before updating (recommended) [Y/n]" chBck
+			case $chBck in
+				Y|y|Yes|yes|"")
+					echo -e "\n${GREEN}Creating a backup in the 'backup' folder${NC}"
+					mkdir -p backup
+
+					printf " - retrieving the database connection details "
+					HOST=$(getConfigVar 'dbHost' )
```

Comment:
You are correct. We could also search for an environment variable. 

I would rather not use the backup.php because this is part of the application. This would mean that if a release has a blocking bug, people could not install the bugfix and take a backup first.
This script could also easily be adapted to force a reinstallation of the latest version if certain files became corrupted.

So I would rather keep the update/backup part separated from the app. If no username/password/database is found, I could ask for input.

Reply:
So I first check for the environment variable and if not found, check if it is in the configuration.php (same order as we do in configEnvironmentHelper).
If it still is not found (for example, when there is a corrupt install of some sort), we can enter it manually.